File: app/src/main/java/wannabit/io/cosmostaion/network/res/ResOkHistory.java
package wannabit.io.cosmostaion.network.res;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class ResOkHistory {

    @SerializedName("code")
    public String code;

    @SerializedName("msg")
    public String msg;

    @SerializedName("data")
    public ArrayList<Data> data;

    public class Data {
        @SerializedName("transactionLists")
        public ArrayList<transactionData> transactionLists;

        public class transactionData {
            @SerializedName("txId")
            public String txId;

            @SerializedName("blockHash")
            public String blockHash;

            @SerializedName("height")
            public String height;

            @SerializedName("transactionTime")
            public String transactionTime;

            @SerializedName("from")
            public String from;

            @SerializedName("to")
            public String to;

            @SerializedName("amount")
            public String amount;

            @SerializedName("transactionSymbol")
            public String transactionSymbol;

            @SerializedName("txFee")
            public String txFee;

            @SerializedName("state")
            public String state;
        }
    }
}

File: app/src/main/java/wannabit/io/cosmostaion/network/res/ResOkHistory.kt
package wannabit.io.cosmostaion.network.res

import com.google.gson.annotations.SerializedName

data class ResOkHistory(
    @SerializedName("code")
    var code: Int,

    @SerializedName("msg")
    var message: String,

    @SerializedName("detail_msg")
    var detailMessage: String,

    @SerializedName("data")
    var data: ResOkHistoryData
)

class ResOkHistoryData(
    @SerializedName("hits")
    var hits: List<ResOkHistoryHit>
)

class ResOkHistoryHit(
    @SerializedName("hash")
    var hash: String,

    @SerializedName("blocktime")
    var blocktime: Long,

    @SerializedName("blockHash")
    var blockHash: String,

    @SerializedName("blockHeight")
    var blockHeight: Long,

    @SerializedName("transactionData")
    var transactionDatas: List<ResOkHistoryTransactionData>,

    @SerializedName("blockTimeU0")
    var blockTimeU0: Long
)

class ResOkHistoryTransactionData(
    @SerializedName("type")
    var type: String
)
